libr/io/p/io_r2k.c

```diff
@@ -313,20 +347,121 @@ static int WriteKernelMemory_linux (RIODesc *iodesc, ut64 address, const ut8 *bu
 		}
 
 		free (data.buff);
-		return ret;
+	} else if (!buf) {
+		eprintf ("Invalid input buffer.\n");
 	} else {
 		eprintf ("IOCTL device not initialized.\n");
-		return -1;
 	}
+	return ret;
 }
+
+static int run_ioctl_command (RIODesc *iodesc, const char *buf) {
+	int ret, inphex, ioctl_n;
+	ut64 pid, addr, len;
+	ut8 *databuf = NULL;
+	buf = r_str_ichr ((char *)buf, ' ');
+
+	switch(*buf) {
+	case 'r':
+		{
+			if (buf[1] == 'l') {
+				//read linear address
+				//=! rl addr len
+				pid = 0;
+				addr = getvalue (buf, 1);
+				len = getvalue (buf, 2);
+			    ioctl_n = IOCTL_READ_KERNEL_MEMORY;
+			} else if (buf[1] == 'p') {
+				//read process address
+				//=! rp pid address len
+				pid = getvalue (buf, 1);
+				addr = getvalue (buf, 2);
+				len = getvalue (buf, 3);
+			    ioctl_n = IOCTL_READ_PROCESS_ADDR;
+			} else if (buf[1] == 'P') {
+				//read physical address
+				//=! rP address len
+				pid = 0;
+				addr = getvalue (buf, 1);
```

Comment:
Yup. I realized it earlier. Haven't updated the PR yet :). Thanks for review though.